202308-quarto-websites/quarto-websites-notes.qmd

```diff
@@ -0,0 +1,190 @@
+    -   Compare file contents to website preview
+    -   Creates files in `_sites` folder
+        -   Do not manually edit these
+    -   Update after changing files by rendering again
+    -   Can also do from command line with `quarto preview`
+
+### Deploy website
+
+-   Do this to make website available online to others
+-   Straightforward to do
+-   Many possible platforms, some free and some paid
+    -   Posit Connect
+        -   We have this at UA - primarily for research
+    -   Netlify
+        -   Can get custom domain
+        -   Can make private sites available to authenticated users
+    -   GitHub pages
+        -   Need GH account + git knowledge
+    -   Quarto Pub
+        -   Easy and free
+        -   Only public content
+        -   Need an account
+-   Publish example website to Quarto Pub
+    -   No point-and-click way to do this (yet)
+    -   In RStudio "Terminal" tab
+    -   Should already be in correct directory because of RStudio project
+    -   Type in and execute `quarto publish quarto-pub`
+    -   Need to be logged into Quarto Pub in browser
+    -   Authorize with `Y` in Terminal, then hit "Allow" button in browser
+-   Deployed website on Quarto Pub
+    -   Link to website is at top
+    -   Can change slug (end part of URL) in right hand pane
+-   Update deployed website
+    -   Make a change (website title in `index.qmd` & `_quarto.yml` to name)
+    -   Run `quarto publish quarto-pub` command again
+-   Suggested workflow
+    -   Make a bunch of changes locally, then publish
+
+### Customizing website
+
+-   Lots of customization options
+    -   Add a page
+        -   Create new `.qmd` file, adding a title and content
+        -   Add file under `navbar` in `_quarto.yml`
+        -   Name of page on website comes from yml
+    -   Use template for "About" page
+        -   [About templates](https://quarto.org/docs/websites/website-about.html#templates)
+        -   Add `image` and `about` to `index.qmd`
+        -   Add image file to main-level project folder
+        -   Turn "Home" into "About"
+        -   Add social media links with `links`
+        -   Follow yml formatting correctly
+    -   Change theme
+        -   Built-in options from [Bootswatch](https://bootswatch.com/)
+        -   Update at `theme` in `_quarto.yml`
+    -   Add logo to navbar
+        -   Find and save image file in main-level project folder
+        -   Add `logo` under `navbar`
+    -   Add blog with posts
+        -   Nice for sharing code examples or outputs
+        -   Not actually going to go through this
+        -   [How to](https://samanthacsik.github.io/posts/2022-10-24-quarto-blogs/)
+    -   Do more fancy things with CSS
+        -   Beyond scope of this workshop
+-   Get inspiration from some great Quarto websites!
+    -   [Eric](https://ericrscott.com/)
+    -   [Maya Gans](https://maya.rbind.io/)
+    -   [Heidi Steiner](https://heidiesteiner.netlify.app/)
```

Comment:
If you send this out in a followup email it'd be useful to link directly the corresponding github repos as well